src/Query/LaravelReadQuery.php

```diff
@@ -105,19 +105,22 @@ public function getResourceSet(
             assert($numValues == count($segments));
 
             for ($i = 0; $i < $numValues; $i++) {
```

Comment:
This is setting up for application in next foreach look - where primaryField > tail end charlie's primary value OR where primaryField == TEC's primary value and secondaryField > TEC's secondary value OR where both primary and secondary fields are equal to corresponding TEC values and tertiaryField > TEC's tertiary value, etc